tests/test_random_variables/test_normal.py

```diff
@@ -413,6 +415,56 @@ def test_transpose(self):
 
         self.assertArrayEqual(dist_t_sample, dist_sample)
 
+    def test_cov_cholesky_cov_cholesky_not_passed(self):
+        """No cov_cholesky is passed in init.
+
+        In this case, the "is_precomputed" flag is False, a cov_cholesky
+        is computed on demand, but can also be computed manually with
+        any damping factor.
+        """
+        mean, cov = self.params
+        rv = rvs.Normal(mean, cov)
+
+        with self.subTest("No Cholesky precomputed"):
+            self.assertFalse(rv.cov_cholesky_is_precomputed)
+
+        with self.subTest("Without explicit damping"):
+            # damping factor 1e-12 does not mess up this test
+            self.assertAllClose(rv.cov_cholesky, np.sqrt(rv.cov))
+
+        with self.subTest("With explicit damping"):
+            # Sanity check: usual cov_cholesky does not work
+            with self.assertRaises(AssertionError):
+                self.assertAllClose(rv.cov_cholesky, np.sqrt(rv.cov + 10.0))
+
+            rv.precompute_cov_cholesky(damping_factor=10.0)
+            self.assertAllClose(rv.cov_cholesky, np.sqrt(rv.cov + 10.0))
+
+    def test_cov_cholesky_cov_cholesky_passed(self):
+        """A value for cov_cholesky is passed in init.
+
+        In this case, the "is_precomputed" flag is True, the
+        cov_cholesky returns the argument that has been passed, but
+        (p)recomputing overwrites the argument with a new factor.
+        """
+        mean, cov = self.params
+
+        # explicitly not the correct cov_cholesky for test reasons
+        cov_cholesky = np.random.rand()
+
+        rv = rvs.Normal(mean, cov, cov_cholesky=cov_cholesky)
+
+        with self.subTest("Cholesky precomputed"):
+            self.assertTrue(rv.cov_cholesky_is_precomputed)
+
+        with self.subTest("Returns correct cov_cholesky"):
+            self.assertAllClose(rv.cov_cholesky, cov_cholesky)
+
+        with self.subTest("self.precompute overwrites with true factor"):
+            # Sanity check: usual cov_cholesky does not work
+            rv.precompute_cov_cholesky()
+            self.assertAllClose(rv.cov_cholesky, np.sqrt(rv.cov))
```

Comment:
I suppose this "overwrite factor even a Cholesky factor has been provided initially" is an API decision that one can argue about. 
Let me know your thoughts please.

Reply:
Personally, I would not allow overwriting the provided Cholesky factor. What are your thoughts, @JonathanWenger?

Reply:
I agree. Random variables should be as immutable as possible, barring their random seed imo.

Reply:
Alright. Then I'd suggest throwing an error (alternative would be a `pass`, which is a bit odd?!).

Reply:
ummmm which one? :D TypeError? RuntimeError? (a list is in https://docs.python.org/3/library/exceptions.html)

Reply:
I think I would just throw a plain `Exception` since none of the standard exceptions really fit here